business_objects/monitor.py

```diff
@@ -0,0 +1,424 @@
+) -> None:
+    query = f"""
+    UPDATE record_tokenization_task
+    SET state = '{enums.TokenizerTask.STATE_FAILED.value}'
+    WHERE state = '{enums.TokenizerTask.STATE_IN_PROGRESS.value}'
+    """
+
+    if project_id:
+        query = (
+            query
+            + f"""
+        AND project_id = '{project_id}'
+        """
+        )
+    if task_id:
+        query = (
+            query
+            + f"""
+        AND id = '{task_id}'
+        """
+        )
+    print(query)
+    general.execute(query)
+    general.flush_or_commit(with_commit)
+
+
+def set_embedding_to_failed(
+    project_id: str = None, embedding_id: str = None, with_commit: bool = False
+) -> None:
+    query = f"""
+    UPDATE embedding
+    SET state = '{enums.EmbeddingState.FAILED.value}'
+    WHERE state = '{enums.EmbeddingState.ENCODING.value}' OR state = '{enums.EmbeddingState.WAITING.value}' OR state = '{enums.EmbeddingState.INITIALIZING.value}'
+    """
+
+    if project_id:
+        query = (
+            query
+            + f"""
+        AND project_id = '{project_id}'
+        """
+        )
+    if embedding_id:
+        query = (
+            query
+            + f"""
+        AND id = '{embedding_id}'
+        """
+        )
+    general.execute(query)
+    general.flush_or_commit(with_commit)
+
+
+def set_weak_supervision_to_failed(
+    project_id: str = None, task_id: str = None, with_commit: bool = False
+) -> None:
+    query = f"""
+    UPDATE weak_supervision_task
+    SET state = '{enums.PayloadState.FAILED.value}'
+    WHERE state = '{enums.PayloadState.STARTED.value}'
+    """
+
+    if project_id:
+        query = (
+            query
+            + f"""
+        AND project_id = '{project_id}'
+        """
+        )
+    if task_id:
+        query = (
+            query
+            + f"""
+        AND id = '{task_id}'
+        """
+        )
+    general.execute(query)
+    general.flush_or_commit(with_commit)
+
+
+def set_upload_task_to_failed(
+    project_id: str = None, task_id: str = None, with_commit: bool = False
+) -> None:
+    query = f"""
+    UPDATE upload_task
+    SET state = '{enums.UploadStates.ERROR.value}'
+    WHERE state = '{enums.UploadStates.IN_PROGRESS.value}' OR state = '{enums.UploadStates.PENDING.value}' OR state = '{enums.UploadStates.PREPARED.value}' OR state = '{enums.UploadStates.WAITING.value}'
+    """
+
+    if project_id:
+        query = (
+            query
+            + f"""
+        AND project_id = '{project_id}'
+        """
+        )
+    if task_id:
+        query = (
+            query
+            + f"""
+        AND id = '{task_id}'
+        """
+        )
+    general.execute(query)
+    general.flush_or_commit(with_commit)
+
+
+def __select_running_information_source_payloads(
+    project_id: str = None, only_running: bool = False
+) -> str:
+    query = f"""
+    SELECT id, 'information_source' task_type, state, project_id, created_by
```

Comment:
should we maybe use the tablenames enum?

Reply:
Done